### ISSUE TITLE
`torch_geometric.explain` + built-in `Captum` support

### ISSUE THREAD

**Issue by rusty1s** [pyg-team/pytorch_geometric#3858]
## Roadmap

1. `Explainer`: `ABC` for explainability
2. `GNNExplainer(Explainer)`
3. `PGExplainer(Explainer)`
4. `to_captum` functionality
5. `to_captum` example
6. `evaluate_explainer` functionality. Some approaches to evaluate explainers are mentioned in [paper1](https://proceedings.neurips.cc/paper/2021/hash/99bcfcd754a98ce89cb86f73acc04645-Abstract.html) and [paper2](https://dl.acm.org/doi/abs/10.1145/3447548.3467283?casa_token=I2_R42o5b_EAAAAA%3AxSqT6VoJw2n6Tbck5_CnTU2u0IEbLisldEWrKWY3UFjF76e6MuQp8cbNU0BFsCPRngh1tyEVCa0r4A)
6. Implement/move to own `torch_geometric.explain` subpackage

```python
def to_captum(model: torch.nn.Module, mask_type: str = "edge",
              node_idx: Optional[int] = None) -> torch.nn.Module

...

model(mask, x, edge_index)
```

```python
def forward(self, mask, *args, **kwargs):
    set_masks(model, mask)
    out = self.model(*args, **kawrgs)
    clear_masks(model)
    return out
```

**Comment by wsad1** [comment]
This is a nice idea. I'd be happy to help with this.
#2576 this is an old open WIP PR for PG explainer,  we could modify it to conform to `Explainer`.
Another thing what functions would `Exaplainer` have, I can think of `set_mask`, `clear_mask` and `train_explainer` thats common across PGexplainer and GNNExplainer.

**Comment by rusty1s** [comment]
That's great, thanks for your help. @RBendias will implement the `Explainer` base class, which takes care of settings mask and providing unified visualizations. After the PR has landed, we can try to map #2576 to respect the new interface.

**Comment by RBendias** [comment]
@wsad1 I already looked into your PR for PGExplainer a few weeks ago & also modified it to the current GNNExplainer state but didn't push it to your PR yet. I'm writing my thesis on Explainability for GNN; therefore, your commits helped me a lot already. Thanks!

**Comment by wsad1** [comment]
@RBendias  all the best with you thesis, it sounds great.
Looking forward to your PR.

**Comment by wsad1** [comment]
There seems to be a gap when it come to evaluation of explainers. So as a start we could add the 2 evaluation methods used in this [paper](https://proceedings.neurips.cc/paper/2021/hash/99bcfcd754a98ce89cb86f73acc04645-Abstract.html) to the `explain` module. WDYT?

**Comment by RBendias** [comment]
Thanks for sharing the paper. Indeed, I also see gaps when it comes to the evaluation of those methods. This is another [paper](https://dl.acm.org/doi/abs/10.1145/3447548.3467283?casa_token=I2_R42o5b_EAAAAA%3AxSqT6VoJw2n6Tbck5_CnTU2u0IEbLisldEWrKWY3UFjF76e6MuQp8cbNU0BFsCPRngh1tyEVCa0r4A) that I find very interesting on this subject.

**Comment by wsad1** [comment]
Yeah this looks interesting. I'll add this to the roadmap too then.

**Comment by rusty1s** [comment]
@fork123aniket
